clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -497,6 +528,32 @@ func (w *waitForControlledPodsRunningMeasurement) getObjectKeysAndMaxVersion() (
 	return objectKeys, maxResourceVersion, nil
 }
 
+type wrapperPodStore struct {
+	indexer       cache.Indexer
+	namespace     string
+	labelSelector labels.Selector
+}
+
+func (w *wrapperPodStore) List() []*v1.Pod {
+	objects, err := w.indexer.ByIndex("metadata.namespace", w.namespace)
```

Comment:
I'm wondering if this is sufficient.

Before this PR, we used do filtering server side which saved tons of CPU here.
Now we will be iterating all pods in a namespace every 5s from multiple WaitForPods calls.

Do we have some data about CPU usage of pod running the test after this change? How it changed wrt to old pattern?

I was thinking about using some user-provided indexing key, but your approach is better, if it solves the problem properly.

Reply:
User-provided stuff is always more error prone, so I wanted to avoid that.
And in practice our namespaces are relatively small.

I was relying a bit on the graph that you pasted on the other PR showing that without any indexing it was using ~2 cores. With indexing, it should on average be using ~50x less in our load test...

We should run it against large test, but I'm wondering if we already have any way to automatically grab the information about the test resource usage itself?

Reply:
There is no automatic way. You need to go to the stackdriver of cluster running a prow pod and check cpu usage there (you should have an access). This is how the graph on the other PR was created.

Reply:
You could build a test binary, then run that binary under `/usr/bin/time -v`. That should give you a total of the resource consumption of the test driver itself, with output directly in the log file.

Reply:
The CPU graph for presubmit of this PR:
![image](https://user-images.githubusercontent.com/2559168/166646930-c9a0dab7-c787-4e60-9973-6dae3224392c.png)

You can compare this with graphs in https://github.com/kubernetes/perf-tests/pull/2016#discussion_r843925817

Overall, it looks good to me.